src/Npgsql/NpgsqlTypes/PostgisTypes.cs

```diff
@@ -105,22 +275,38 @@ internal int GetLen(bool includeSRID)
     }
 
     /// <summary>
-    /// Represents an Postgis 2D Point
+    /// PostGIS base Geometry type with specified coordinate system.
     /// </summary>
-    public class PostgisPoint : PostgisGeometry, IEquatable<PostgisPoint>
+    /// <typeparam name="T">Coordinate type for Geometry</typeparam>
+    public abstract class PostgisGeometry<T> : PostgisGeometry where T : Coordinate2D
```

Comment:
Suggestion: we could have PostgisGeometry, PostgisGeometryZ, PostgisGeometryM and PostgisGeometryZM. At the very least, this would allow for more conciseness when creating a geometry collection:

```c#
new PostgisGeometryCollection(new PostgisGeometry3DZ[] {
```

instead of:

```c#
new PostgisGeometryCollection(new PostgisGeometry<Coordinate3DZ>[] {
```

Reply:
Agreed, need to remove nasty syntax. Will be in next push.